### PR TITLE
Fixes #20 Added Support for SNLI Dataset

### LINKED ISSUE #20: [data request] SNLI
* Name of dataset: SNLI
* URL of dataset: https://nlp.stanford.edu/projects/snli/
* License of dataset: Creative Commons Attribution-ShareAlike 4.0 International License
* Short description of dataset and use case(s):
The SNLI corpus (version 1.0) is a collection of 570k human-written English sentence pairs manually labeled for balanced classification with the labels entailment, contradiction, and neutral, supporting the task of natural language inference (NLI), also known as recognizing textual entailment (RTE)

Folks who would also like to see this dataset in `tensorflow/datasets`, please +1/thumbs-up so the developers can know which requests to prioritize.

### PR DESCRIPTION
**Pull request by ParthS007**
Fixes #20 
@Conchylicultor @rsepassi @cyfra @dynamicwebpaige @ChanchalKumarMaji @us Please review and let me know if I have to do any changes. Thanks :)

### CODE REVIEW

File: tensorflow_datasets/text/snli.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        citation=_CITATION,
+    )
+
+  def _vocab_text_gen(self, filepath):
+    for ex in self._generate_examples(filepath):
+      yield " ".join([ex["text"], ex["hypothesis"], ex["judgements"]])
+
+  def _split_generators(self, dl_manager):
+
+    downloaded_dir = dl_manager.download_and_extract(
```
Comment (us) [inline tensorflow_datasets/text/snli.py]: Yo shouldn't be written URL directly here. Define URL above.

File: tensorflow_datasets/url_checksums/snli.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+https://nlp.stanford.edu/projects/snli/snli_1.0.zip 94550081 afb3d70a5af5d8de0d9d81e2637e0fb8c22d1235c2749d83125ca43dab0dbd3e
```
Comment (us) [inline tensorflow_datasets/url_checksums/snli.txt]: Why put checksum in a different file. If I'm not wrong, checksum is already being retrieved and checked while it is being downloaded.

File: tensorflow_datasets/text/snli.py (hunk 1 above)
Comment (Conchylicultor) [reply tensorflow_datasets/text/snli.py]: Honestly, I don't think it matters much. But I agree having them as constant at the top of the file would be consistent with other datasets

File: tensorflow_datasets/url_checksums/snli.txt (hunk 2 above)
Comment (Conchylicultor) [reply tensorflow_datasets/url_checksums/snli.txt]: This file is automatically generated when adding a new dataset.
When users `.download_and_prepare()` a datasets, the urls checksums are compared with this file to ensure the downloaded files are correct.

Comment (ParthS007) [reply tensorflow_datasets/url_checksums/snli.txt]: @us In one of the previous PR, this has been added by @pierrot0 to store the checksum in a different file. Link is [Here](https://github.com/tensorflow/datasets/pull/247)

File: tensorflow_datasets/text/snli.py (hunk 1 above)
Comment (ParthS007) [reply tensorflow_datasets/text/snli.py]: okay @Conchylicultor @us , I will add them in the top. :+1:

File: tensorflow_datasets/url_checksums/snli.txt (hunk 2 above)
Comment (us) [reply tensorflow_datasets/url_checksums/snli.txt]: @ParthS007 okay I missed it thank you!

**Comment by ParthS007** [comment]
@us @Conchylicultor I have made the changes and ran the test. All the tests are passing. Please review and let me know if I have to do any changes. Thanks :)

**Comment by ParthS007** [comment]
@cyfra Please review. Thanks :)

**Comment by ParthS007** [comment]
@Conchylicultor @cyfra @tfds-copybara Please review. Thanks :)

File: tensorflow_datasets/text/snli.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  ]
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "text":
+                tfds.features.Text(
+                    encoder_config=self.builder_config.text_encoder_config),
```
Comment (us) [inline tensorflow_datasets/text/snli.py]: Why didn't you define anything about text_encoder_config.

Comment (ParthS007) [reply tensorflow_datasets/text/snli.py]: Please check `BUILDER_CONFIGS` for the same. Thanks for the review

Comment (us) [reply tensorflow_datasets/text/snli.py]: I thought maybe there could take a few specific parameters about your dataset.

**Comment by ParthS007** [comment]
@cyfra Please review. Thanks :)

**Comment by cyfra** [comment]
Added in #679

**Pull request closed without merging**